lib/components/intent.js

```diff
@@ -649,17 +649,17 @@ Intent.prototype._ensureRegistered = function() {
     if (this.opts.registered) {
         return Promise.resolve("registered=true");
     }
-    var self = this;
-    var userId = self.client.credentials.userId;
-    var localpart = new MatrixUser(userId).localpart;
-    return self.botClient.register(localpart).then(function(res) {
-        self.opts.registered = true;
+    const userId = this.client.credentials.userId;
+    const localpart = new MatrixUser(userId).localpart;
+    return self.botClient.register(localpart).then((res) => {
+        this.opts.registered = true;
         return res;
-    }, function(err) {
+    }).catch((err) => {
         if (err.errcode === "M_USER_IN_USE") {
-            self.opts.registered = true;
+            this.opts.registered = true;
             return err;
```

Comment:
If we're rejecting the promise, we should probably suppress this error or let it fall through to the throw below, whichever is best.

Reply:
Supression works for me.